src/vaults/VaultRouter.sol

```diff
@@ -102,13 +102,29 @@ contract VaultRouter is Auth, Multicall, Recoverable, IVaultRouter {
 
         VaultDetails memory vaultDetails = poolManager.vaultDetails(vault);
         if (owner == address(this)) {
-            _approveMax(vaultDetails.asset, vaultDetails.tokenId, address(vault));
+            _approveMax(vaultDetails.asset, address(vault));
         }
 
         _pay();
         vault.requestDeposit(amount, controller, owner);
     }
 
+    /// @inheritdoc IVaultRouter
+    function deposit(BaseSyncDepositVault vault, uint256 assets, address receiver, address owner)
```

Comment:
We didn't support this in the router before. By doing that, we can 1) use permits for deposits, 2) automatically pay, 3) use ERC20 wrapping.

Reply:
Fix related to allowances in https://github.com/centrifuge/protocol-v3/pull/322/files#diff-b718607f380b901292cff484622241096a53434bcdc641336afa997895920c24R123